### PR TITLE
[superuser] visualize data in the database

### PR DESCRIPTION
**Pull request by seshness**
This is the data in our database! http://d.pr/i/AOpK

### CODE REVIEW

File: app/controllers/superuser_controller.rb [hunk 1]
```diff
@@ -2,4 +2,27 @@ class SuperuserController < ApplicationController
   def index
     @departments = Department.all
   end
+
+  def timelines
+    encounters = Encounter.where('patient_id IN (SELECT id FROM patients WHERE id IN ' \
+                           '(SELECT patient_id FROM encounters WHERE ' \
+                           'encounters.end_time IS NULL OR encounters.end_time >= ? ))',
+                           12.hours.ago).order('end_time ASC').to_a
```
Comment (danwang) [inline app/controllers/superuser_controller.rb]: what the fuck

File: app/controllers/superuser_controller.rb [hunk 2]
```diff
@@ -2,4 +2,27 @@ class SuperuserController < ApplicationController
...
+
+    result = []
+    encounters.group_by(&:patient_id).map do |patient_id, encounters|
```
Comment (danwang) [inline app/controllers/superuser_controller.rb:14]: What is `&:patient_id`?

File: app/controllers/superuser_controller.rb [hunk 3]
```diff
@@ -2,4 +2,27 @@ class SuperuserController < ApplicationController
...
+      result << {
+        'name' => Patient.find(patient_id).name,
+        'stages' => encounters.map do |e|
+          {
+            'name' => e.department.name,
+            'start' => e.start_time,
+            'end' => e.end_time
```
Comment (danwang) [inline app/controllers/superuser_controller.rb]: Can we use `:name`, `:start`, and `:end`?

File: app/controllers/superuser_controller.rb (hunk 2 above)
Comment (seshness) [inline app/controllers/superuser_controller.rb:14]: See http://api.rubyonrails.org/classes/Enumerable.html#method-i-group_by

File: app/controllers/superuser_controller.rb (hunk 1 above)
Comment (danwang) [inline app/controllers/superuser_controller.rb]:
```
ending_encounters = Encounter.where('end_time IS NULL OR end_time >= :end_time', {:end_time => 12.hours.ago})
patients = ending_encounters.map {|e| e.patient}.uniq{|p| p.id}
encounters = patients.map {|p| p.encounters}.inject{|a, b| a + b}
```

Or something like that

At least not the SQL

**Comment by danwang** [comment]
可以。

**Pull request merged**